modules/mcuboot/hooks/nrf53_hooks.c

```diff
@@ -19,9 +19,51 @@
 #define NET_CORE_VIRTUAL_PRIMARY_SLOT 3
 
 #include <dfu/pcd.h>
+#include <fw_info_bare.h>
 
-int boot_read_image_header_hook(int img_index, int slot,
-		struct image_header *img_head)
+/**
+ * Compare image version numbers
+ * @retval -1           If ver1 is less than ver2.
+ * @retval 0            If the image version numbers are equal.
+ * @retval 1            If ver1 is greater than ver2.
+ */
+int boot_version_cmp_net(const struct flash_area *fap, struct image_header *hdr)
+{
+	int err;
+	const struct fw_info *firmware_info;
+	uint32_t version = 0;
+	uint32_t read_flash_page[4096];
```

Comment:
Should this buffer size be configurable? Or at least come from a read-only kconfig option?

Reply:
I wasn't able to reply to the above comment:

> But why do we need to add it now?

Because you could end up reading the version and it failed, try to start the application. This would fail as the network core would be in secure while the application expects non-secure.

> Should this buffer size be configurable? Or at least come from a read-only kconfig option?

I'll look into adding it as a read-only konfig. This buffer just needs to be large enough to find the firmware version in the network core.